### ISSUE TITLE
How to confirm the IP through which the link is connected to the server when the client has several IPs

### ISSUE THREAD

**Issue by peguine** [eclipse-paho/paho.mqtt.cpp#263]
One requirement is that when the client goes online, it needs to report the client IP, but there are multiple network cards on the client side, and the connection to the server may be through 4G or wired. How to determine which way to connect successfully. No related interface found in API documentation。

**Comment by fpagliughi** [comment]
Yeah, there's no way to do this through the API. That's not something I've generally seen in client libraries for different protocols, other than maybe exposing the underlying socket handle, and allowing the application make some target/OS-specific calls with it.

And it wouldn't be too hard to do right now, since this library and most MQTT implementations ride (ultimately) on top of TCP. But that requirement was removed from the standard, and it is now allowed to be implemented on top of any "ordered, lossless, [bidirectional] stream of bytes".

**Comment by fpagliughi** [comment]
Duplicate of #237
